deploy/charts/rook-ceph/templates/clusterrole.yaml

```diff
@@ -116,18 +116,26 @@ rules:
   - create
   - update
   - delete
-- apiGroups:
-  - ceph.rook.io
-  resources:
-  - "*"
-  verbs:
-  - "*"
-- apiGroups:
-  - rook.io
-  resources:
-  - "*"
-  verbs:
-  - "*"
+# The Rook operator must be able to watch all ceph.rook.io resources to reconcile them.
+- apiGroups: ["ceph.rook.io"]
+  resources: ["*"]
+  verbs: ["get", "list", "watch"]
```

Comment:
Ideally, (when/if k8s makes finalizers a fully-fledged sub-resource), Rook will only have read permissions on the user-controlled portions of the resource, and it will only have write access to finalizer/status subresources. However, in the current implementation, we still need update access for finalizers.

It's hard to explain, but the comment here does the best job I can about why the `finalizer` subresource may be needed. This showed up recently with OBCs and not being able to set OBCs as owners of ConfigMaps/Secrets.
```
# The "finalizers" permission may need to be strictly given for clusters where
# OwnerReferencesPermissionEnforcement is enabled so that Rook can set blockOwnerDeletion on
# resources owned by Rook CRs. See more:
# https://kubernetes.io/docs/reference/access-authn-authz/_print/#ownerreferencespermissionenforcement
```

Reply:
Until we don't need "update" on the full CRDs, what about consolidating to something like this? 
```yaml
- apiGroups: ["ceph.rook.io"]
  resources: <List all CRDs>
  # Later we can remove "update" after k8s changes the finalizers, etc...
  verbs: ["get", "list", "watch", "update"]
- apiGroups: ["ceph.rook.io"]
  resources: ["*/status", "*/finalizers"]
  verbs: ["update"]
```